### PR TITLE
[python-api] add xbmcgui.Dialog().contextmenu()

### PR DESCRIPTION
**Pull request by phil65**
When workin with WindowXML objects, creating some sort of contextmenu is quite painful. Both me and @ronie workarounded that in some ways (by either using DialogSelect() or by using a custom script window XML)
This should allow to get rid of these workarounds.

@tamland

EDIT: renamed to contextmenu() which suits better to overall naming.

### CODE REVIEW

File: xbmc/interfaces/legacy/Dialog.cpp [hunk 1]
```diff
@@ -84,6 +85,22 @@ namespace XBMCAddon
...
+      CGUIDialogContextMenu* pDialog= (CGUIDialogContextMenu*)g_windowManager.GetWindow(WINDOW_DIALOG_CONTEXT_MENU);
+      if (pDialog == NULL)
+        throw WindowException("Error: Window is NULL, this is not possible :-)");
+
+      CContextButtons choices;
+      for(unsigned int i = 0; i < list.size(); i++)
+      {
+        choices.Add(i, list[i]);
+      }
+      return pDialog->ShowAndGetChoice(choices);
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.cpp]: Use `Show`. This one is only for the old menus using button ids.

File: xbmc/interfaces/legacy/Dialog.h [hunk 2]
```diff
@@ -78,6 +78,19 @@ namespace XBMCAddon
...
       /**
+       * contextmenu(list) -- Show a context menu.\n
+       * \n
+       * list           : string list - list of items.\n
+       * \n
+       * *Note, Returns the position of the highlighted item as an integer (-1 if cancelled).\n
+       * \n
+       * example:\n
+       *   - dialog = xbmcgui.Dialog()\n
+       *   - ret = dialog.context(['Option #1', 'Option #2', 'Option #3'])\n\n
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.h]: "contextmenu"

**Comment by phil65** [comment]
updated. also fixed some more typos in docs.

**Comment by ronie** [comment]
while testing this, i noticed there's a positioning issue (not related to this PR).

if the script xml is coded in 1080, but the script is run in a 720p skin,
the context menu isn't centered correctly.

for example, this is what happens in confluence (when using the script provided xml):
contextmenu

**Comment by phil65** [comment]
yup, can confirm that.
I guess something is wrong here? https://github.com/phil65/xbmc/blob/python_context_menu/xbmc/dialogs/GUIDialogContextMenu.cpp#L685-L687

**Comment by phil65** [comment]
since that bug is not related to this PR:
jenkins build this please

**Pull request merged**